go/vt/vtorc/inst/instance_dao.go

```diff
@@ -88,7 +88,12 @@ func initializeInstanceDao() {
 func ExecDBWriteFunc(f func() error) error {
 	m := query.NewMetric()
 
-	instanceWriteChan <- true
+	ctx, cancel := context.WithTimeout(context.Background(), maxBackendOpTime)
+	defer cancel()
+
+	if err := instanceWriteSem.Acquire(ctx, 1); err != nil {
+		return err
```

Comment:
We didn't have a test before, but maybe its worth adding one to check that these concurrency controls are respected.

Reply:
@GuptaManan100 sounds good 👍 

I've just been optimizing code as I come across it lately and only now I see `ExecDBWriteFunc` is used only in a handful of write operations and `instanceReadSem` is never used

So some more cleanup is needed. Do we want all reads/writes to respect a limit? And considering a majority of reads/writes in VTOrc aren't respecting a limit, do we need one 🤔

Reply:
This is fine too, but if we want to make all the read and write calls acqurie this lock too, then that would technically be more correct.

Reply:
@GuptaManan100 I'm torn, I suspect this semaphore made more sense when the backend was remote and not in-memory

For sqlite3 there is a single writer so I think it already will constrain concurrency, so potentially only a read limit will be useful 🤔 

I merged this PR too soon when I saw it had 2 x reviews, but I can follow up with whatever we decide